src/Datadog.Trace/Abstractions/IEvent.cs

```diff
@@ -0,0 +1,11 @@
+// <copyright file="IEvent.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+namespace Datadog.Trace.Abstractions
+{
+    internal interface IEvent
```

Comment:
The idea was this and our version of AgentWriter might be shared at some point, we plan to move MsgPack at some point, so that will probably make it a little easier.

However, I agree it's probably better to make the separation clear for now, should be fixed in 6cd73399.